### PR TITLE
account import update

### PR DESCRIPTION
**Pull request by bkendall**
### Description

- Removing the dependency on JSONStream. using `stream-json` now, along with `stream-chain`. It's not exactly intuitive, but it works.
- Updates the csv streaming dependency to be `csv-parse`, which has been updated in the past 4 years and provides types for typescript.
- Rewrites import into typescript.

### Scenarios Tested

import and export of csv and json files.

### CODE REVIEW

File: src/commands/auth-import.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  .before(requirePermissions, ["firebaseauth.users.create", "firebaseauth.users.update"])
+  .action(async (dataFile, options) => {
+    const projectId = needProjectId(options);
+    const checkRes = validateOptions(options);
+    if (!checkRes.valid) {
+      return checkRes;
+    }
+    const hashOptions = checkRes;
+
+    if (!_.endsWith(dataFile, ".csv") && !_.endsWith(dataFile, ".json")) {
```
Comment (joehan) [inline src/commands/auth-import.ts:56]: Prefer native `dataFile.endsWith()` + getting rid of lodash from this file

File: src/commands/auth-import.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  .option("--parallelization <parallelization>", "specify the parallelization for standard scrypt.")
+  .option("--block-size <blockSize>", "specify the block size (normally is 8) for standard scrypt.")
+  .option("--dk-len <dkLen>", "specify derived key length for standard scrypt.")
+  .option(
+    "--hash-input-order <hashInputOrder>",
+    "specify the order of password and salt. Possible values are SALT_FIRST and PASSWORD_FIRST. " +
+      "MD5, SHA1, SHA256, SHA512, HMAC_MD5, HMAC_SHA1, HMAC_SHA256, HMAC_SHA512 support this flag."
+  )
+  .before(requirePermissions, ["firebaseauth.users.create", "firebaseauth.users.update"])
+  .action(async (dataFile, options) => {
```
Comment (joehan) [inline src/commands/auth-import.ts:48]: Can we mkae this `dataFile: string, options: Options` so some of these linter warnings go away?

File: src/commands/auth-import.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            let record: string[] = [];
+            while ((record = parser.read()) !== null) {
+              counter++;
+              const trimmed = record.map((s) => {
+                const str = s.trim().replace(/^["|'](.*)["|']$/, "$1");
+                return str === "" ? undefined : str;
+              });
+              const user = transArrayToUser(trimmed);
+              const err = _.get(user, "error");
+              if (err) {
```
Comment (joehan) [inline src/commands/auth-import.ts:82]: We can get rid of the previous line and just make this `if (user.error)`

File: src/commands/auth-import.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    } else {
+      userListArr = await new Promise<any[]>((resolve, reject) => {
+        const pipeline = new Chain([
+          Pick.withParser({ filter: /^users$/ }),
+          StreamArray.streamArray(),
+          ({ value }) => {
+            counter++;
+            const user = validateUserJson(value);
+            const err = _.get(user, "error");
+            if (err) {
```
Comment (joehan) [inline src/commands/auth-import.ts:113]: Same as above - prefer getting rid of previous line and making this `if (user.error)`

**Review by joehan — APPROVED** [review]
Some nits, Chain is a bit hariy, and I think we can get rid of lodash while you're here, but this is wayyyy better than what we had before already

File: src/commands/auth-import.ts (hunk 1 above)
Comment (bkendall) [reply src/commands/auth-import.ts:56]: Done.

File: src/commands/auth-import.ts (hunk 2 above)
Comment (bkendall) [reply src/commands/auth-import.ts:48]: Done.

File: src/commands/auth-import.ts (hunk 3 above)
Comment (bkendall) [reply src/commands/auth-import.ts:82]: Sure.

**Pull request merged**